### PR TITLE
bump containerd to v1.6.15 and k8s.io/* to v0.26.1

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

Prerequisite for https://github.com/DataDog/datadog-agent/pull/14176:
Notes:
- the new `verbose` arguments seem to do nothing (example: https://github.com/kubernetes/cri-api/blob/v0.25.3/pkg/apis/testing/fake_runtime_service.go#L493)
- the fake `GetContainerEvents` does nothing either (so its safe to give it a nil chan). I was unable to pass it something better
- this PR also fixes what looks like a typo for `RemovePodSandbox`

This PR bumps:
- containerd to v1.6.15
- k8s/* to v0.26.1
- remove the replace on apiserver since OTel libs have been upgraded upstream

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -204,7 +204,7 @@ require (
 	k8s.io/apiserver v0.23.8
 	k8s.io/autoscaler/vertical-pod-autoscaler v0.10.0
 	k8s.io/client-go v0.23.8
-	k8s.io/cri-api v0.23.8
+	k8s.io/cri-api v0.25.0
```
Comment (vboulineau) [inline go.mod:207]: Ideally, if you bump one, you should bump all aligned `k8s.io` version (even though I believe `cri-api` is mostly standalone)

Comment (paulcacheux) [reply go.mod:207]: Ah you're right, I hoped that my change would not be that big

**Comment by paulcacheux** [comment]
Alright, so this will be hard for now because of the fork with no-otel situation. It seems the situation will be fixed with https://github.com/kubernetes/apiserver/commit/56ec64dfeb3fa415a6bcd5c4eccb6044a06b91af, that should be available in v0.26.0. I will come back to this when v0.26.0 is released

File: go.mod [hunk 2]
```diff
@@ -87,7 +87,7 @@ require (
 	github.com/clbanning/mxj v1.8.4
 	github.com/cloudfoundry-community/go-cfclient v0.0.0-20210621174645-7773f7e22665
 	github.com/containerd/cgroups v1.0.4
-	github.com/containerd/containerd v1.6.8
+	github.com/containerd/containerd v1.6.9
```
Comment (olivielpeau) [inline go.mod:90]: Could you bump this to 1.6.12 now that it's released? (to include a CVE fix as well)

**Review by olivielpeau — COMMENTED** [review]
Thanks for looking into this @paulcacheux , it seems like k8s.io/apiserver 0.26.0 is now out, so could you pick this up again?

File: go.mod (hunk 2 above)
Comment (paulcacheux) [reply go.mod:90]: yes let me take a look

Comment (paulcacheux) [reply go.mod]: cc @olivielpeau @vboulineau

Comment (olivielpeau) [reply go.mod]: @paulcacheux sorry for only responding now, but the Agent uses Go 1.19 now on `main` (in https://github.com/DataDog/datadog-agent/pull/14930), could you pick this up again?

**Comment by vboulineau** [comment]
@paulcacheux Unfortunately we cannot go forward with this PR as-is, see: https://github.com/DataDog/datadog-agent/pull/15139/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R585

containerd is already bumped there as well.

**Comment by paulcacheux** [comment]
That's a great news, it's a real fight to bump all those deps. I will close this PR then

**Pull request closed without merging**